src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[0];
+        values = (V[]) new Object[0];
+    }
+
     @Override
     public void put(K key, V value) {
+        int keyIndex = getKeyIndex(key);
+        if (keyIndex == -1) {
+            keys = increaseArraySizeByOne(keys);
+            values = increaseArraySizeByOne(values);
+            keyIndex = keys.length - 1;
+        }
+        keys[keyIndex] = key;
+        values[keyIndex] = value;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < keys.length; i++) {
+            if (keys[i] == key || keys[i] != null && keys[i].equals(key)) {
```

Comment:
Let's extract it to a separate method to remove code duplication.
```suggestion
            if (keysAreEqual(...) {
```